website/source/guides/secret-mgmt/db-root-rotation.html.md

```diff
@@ -0,0 +1,409 @@
+   "auth": null
+}
+
+# Make sure that you can connect to the database using the Vault generated credentials
+$ psql -h postgres.host.address -p 5432 \
+       -U v-root-readonly-x6q809467q98yp4yx4z4-1525378026e postgres
+Password for user v-root-readonly-x6q809467q98yp4yx4z4-1525378026:
+
+postgres=> \du
+Role name                                       |                         Attributes                         | Member of
+------------------------------------------------+------------------------------------------------------------+----------
+postgres                                        | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
+v-root-readonly-x6q809467q98yp4yx4z4-1525378026 | Password valid until 2018-05-03 21:07:11+00                | {}
+v-root-readonly-x7v65y1xuprzxv9vpt80-1525378873 | Password valid until 2018-05-03 21:21:18+00                | {}
+
+postgres=> \q
+```
+<br>
+
+This confirms that the Vault successfully connected to your PostgreSQL server
+and created a new user based on the privilege defined by `readonly.sql`.
+
+> The user credentials generated by the Vault has a limited TTL based on your
+configuration (`default_ttl`). In addition, you can revoke them if you suspect
+that it was compromised.
+
+**NOTE:** The `pg_hba.conf` is configured to prompt for password to
+demonstrate this feature.
+
+```plaintext
+TYPE  DATABASE        USER            ADDRESS                 METHOD
+
+local   all             all                                   trust
+host    all             all           0.0.0.0/0               password
+```
+
+
+### <a name="step4"></a>Step 4: Rotate the root credentials
+
+Vault provides an API endpoint to easily rotate the root database credentials.
+
+#### CLI command
+
+```plaintext
+$ vault write -force database/rotate-root/postgresql
+```
+
+#### API call using cURL
+
+```plaintext
+$ curl --header "X-Vault-Token: ..." \
+       --request POST \
+       http://127.0.0.1:8200/v1/database/rotate-root/postgresql
+```
+
+This is all you need to do.
+
+---
+<br>
+
+Repeat [Step 3](#step3) to verify that Vault continues to generate database
+credentials after the root credential rotation.
+
+To verify that the root credential was rotated:
+
+```plaintext
+$ psql -h postgres.host.address -p 5432 -U root postgres
+Password for user root:
+```
+
+Entering the initial password (e.g. `rootpassword`) will ***not*** work since
+the password was rotated by the Vault.
+
+You can invoke the **`database/rotate-root/:name`** endpoint periodically to
+secure the root credential.
+
+
+~> **NOTE:** Once the root password was rotated, only the Vault knows the new
+password. This is the same for all root database credentials given to Vault.
+Therefore, you should create a separate superuser dedicated to the Vault usage
+and not used for other purposes.  
+
+
+## Next steps
+
+In this guide, you learned how to rotate the root database credentials.
+
+Read the [Java Sample App using Spring Cloud
```

Comment:
I'm not sure how we are doing the next steps in the guides sections but this doesn't seem related.

Reply:
I couldn't think of the next step. :(   The Java app guide leverages both database secret engine and EaaS features of Vault, I linked it as a working example of the DB secret engine.  But you're right that it has nothing to do with DB root credential rotation.  I'll think of a better one to link.